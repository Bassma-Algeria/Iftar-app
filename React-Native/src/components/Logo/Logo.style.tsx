import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  moon: {
    position: 'absolute',
    top: -5,
    left: 3,
    width: 22,
    height: 22,
  },
});

export {styles};
