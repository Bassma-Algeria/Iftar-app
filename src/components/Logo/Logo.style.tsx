import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    alignItems: 'baseline',
  },

  moon: {
    position: 'absolute',
    top: -3,
    left: 4,
    width: 20,
    height: 20,
  },
});

export {styles};
