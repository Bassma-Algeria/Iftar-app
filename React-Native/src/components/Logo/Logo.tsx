import React from 'react';
import {View, Animated} from 'react-native';

import {styles} from './Logo.style';

import {ICONS} from '../../utils/constants/Icons';

import {Header} from '../Header/Header';

interface Props {
  animate?: boolean;
}

// TODO: add the animation to the moon in the logo
const Logo: React.FC<Props> = props => {
  return (
    <View>
      <View style={styles.container}>
        <Header variant="h1">إفطار</Header>
        <Moon {...props} />
      </View>
    </View>
  );
};

const Moon: React.FC<Props> = () => {
  return (
    <Animated.Image
      source={ICONS.moon}
      style={[styles.moon, {transform: [{rotate: '0deg'}]}]}
      resizeMode="contain"
    />
  );
};

export {Logo};
