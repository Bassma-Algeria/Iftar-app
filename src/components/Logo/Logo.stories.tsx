import React from 'react';

import {storiesOf} from '@storybook/react-native';

import {Logo} from './Logo';

const LogoStories = storiesOf('Logo', module);

LogoStories.add('Without animation', () => <Logo />);
// LogoStories.add('With animation', () => <Logo />);
