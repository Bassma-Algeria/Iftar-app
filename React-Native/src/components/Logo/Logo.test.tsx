import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {Logo} from './Logo';

describe('Logo Cpmponent', () => {
  it('should render correclty without animation', () => {
    const tree = renderer.create(<Logo />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
