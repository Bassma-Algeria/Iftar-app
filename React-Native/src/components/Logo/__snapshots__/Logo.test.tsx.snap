// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component should render correclty without animation 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "baseline",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#7C4908",
            "fontFamily": "Cairo-Bold",
            "fontSize": 35,
          },
          undefined,
        ]
      }
    >
      إفطار
    </Text>
    <Image
      collapsable={false}
      nativeID="animatedComponent"
      resizeMode="contain"
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 22,
          "left": 3,
          "position": "absolute",
          "top": -5,
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 22,
        }
      }
    />
  </View>
</View>
`;
